controllers/apply/forms/create-form-router.js

```diff
@@ -92,18 +94,23 @@ module.exports = function(router, formModel) {
         if (isEmpty(formData)) {
             res.redirect(req.baseUrl);
         } else {
-            let successStep = formModel.getSuccessStep();
-            let processSuccess = successStep.processor(formData);
-            processSuccess
+            const successStep = formModel.getSuccessStep();
+            successStep
+                .processor(formData)
                 .then(() => {
-                    res.render('pages/apply/success', {
-                        form: formModel,
-                        success: successStep
+                    // Clear the submission from the session on successfull
+                    unset(req.session, formModel.getSessionProp());
+                    req.session.save(() => {
+                        res.render('pages/apply/success', {
+                            form: formModel,
+                            success: successStep
+                        });
                     });
                 })
                 .catch(err => {
-                    // @TODO handle this
-                    res.send(err);
+                    err.friendlyText = 'There was a problem processing your submission';
```

Comment:
If we get here it's because something in the `processor` failed (eg. it's our fault, not the user's). Maybe we could show some text giving them tips on what to do? eg. their session data should be preserved so we could add a button saying "Try again" or something which sends them back to the review step? No guarantees it'll work a second time but it feels like we could at least offer them some sort of onward step?

Reply:
Good shout